function-sidecar/pkg/dispatcher/grpc/grpc.go

```diff
@@ -52,44 +59,100 @@ func (this *grpcDispatcher) Closed() <-chan struct{} {
 	return this.closed
 }
 
-func (this *grpcDispatcher) handleIncoming() {
-	for {
-		select {
-		case in, open := <-this.input:
-			if open {
-				grpcMessage := toGRPC(in)
-				err := this.stream.Send(grpcMessage)
-				if err != nil {
-					if streamClosureDiagnosed(err) {
-						close(this.closed)
-						return
-					}
-
-					log.Printf("Error sending message to function: %v", err)
-				}
+// A CorrelationStrategy groups together messages that belong to the same "spatial" stream (the "temporal" aspect
+// of grouping is taken care of by a WindowStrategy). Several invocations for different spatial groups may happen concurrently.
+type CorrelationStrategy func(message.Message) interface{}
```

Comment:
Sure. `nil == nil` so while it could clarify things, there is no real need: `nil` is not a special case. It's just one way to do no special partitioning

Reply:
Well, `nil` might indicate a default partition, especially since [`nil` can be a map key](https://play.golang.org/p/eYu-YYKRprb).